### PR TITLE
Fix patching in signing

### PR DESCRIPTION
**Pull request by paweljakubas**
- [x] I have added integration test involving two parties which create wallet via patching...and it passes

### Issue Number
adp-2331

### CODE REVIEW

**Review by piotr-iohk — APPROVED** [review]
Nice!
I've repeated the procedure now also on preview testnet and it also works!
I guess it was either fixed "in the meantime" by some lucky change or I've made a mistake somewhere and there was no issue in the first place...
In any case it is good to have additional integration test that covers this case!

**Comment by paweljakubas** [comment]
bors retry

File: lib/wallet/integration/src/Test/Integration/Scenario/API/Shared/Transactions.hs [hunk 1]
```diff
@@ -1020,3 +932,184 @@ spec = describe "SHARED_TRANSACTIONS" $ do
...
+                [ expectResponseCode HTTP.status200
+                , expectField (traverse . #balance . #available)
+                    (`shouldBe` Quantity faucetUtxoAmt)
+                ]
+
+        verify (fmap (view #wallet) <$> rSharedWal1) balanceExp
+        verify (fmap (view #wallet) <$> rSharedWal2) balanceExp
+
+        wb <- emptyWallet ctx
+        let amt = (minUTxOValue (_mainEra ctx) :: Natural)
```
Comment (jonathanknowles) [inline lib/wallet/integration/src/Test/Integration/Scenario/API/Shared/Transactions.hs:951]: @paweljakubas 

By the way: you can now discover the right minimum UTxO value by:
1. Attempting to create a transaction with an output that has an ada value less than the minimum. (Specifying 1 lovelace should work.)
2. Reading back the required minimum from the error object returned.

Example: https://github.com/input-output-hk/cardano-wallet/blob/2ad0dfa3f789dea56b15ed9cf670852c482bf758/lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs#L240-L249

Hopefully, we should eventually be able to get rid of this `minUTxOValue` function!

**Comment by paweljakubas** [comment]
bors merge

**Pull request merged**